### ISSUE TITLE
timezone.system fails when /etc/localtime is missing on FreeBSD

### ISSUE THREAD

**Issue by ari** [saltstack/salt#44083]
### Description of Issue/Question

Related to #35869 but slightly different. timezone.system fails when /etc/localtime is missing on FreeBSD.

### Setup
FreeBSD 11.1-RELEASE-p1
Salt: 2017.7.1

```
Australia/Sydney:
  timezone.system:
    - name: Australia/Sydney
```

Output:
```
          ID: Australia/Sydney
    Function: timezone.system
      Result: False
     Comment: Unable to compare desired timezone 'Australia/Sydney' to system timezone: Failed to read /etc/localtime to determine current timezone: No such file or directory
     Started: 04:12:12.728588
    Duration: 0.757 ms
```

On the minion

```
ls -al /etc/localtime
ls: /etc/localtime: No such file or directory
```

**Comment by gtmanfred** [comment]
Thanks for reporting, We would love a PR to fix this.

Thanks,
Daniel

**Comment by campbellmc** [comment]
I'm seeing this too.

The problem is in the try block in state `timezone.system` which calls module `timezone.zone_compare` https://github.com/saltstack/salt/blob/develop/salt/states/timezone.py#L61

This is catching a purposely raised exception because `/etc/localtime` doesn't exist.

Here is the result when running the module locally:
```
freebsd # salt-call timezone.zone_compare UTC
Error running 'timezone.zone_compare': Failed to read /etc/localtime to determine current timezone: No such file or directory
```

For a PR, what is the appropriate behaviour/fix here?

Seems a little messy to have to check CommandExecutionError for "No such file or directory" and jump out of the exception to force `timezone.set_zone`?

**Comment by gtmanfred** [comment]
just check if the file exists, and if it doesnt then follow a different code path to create the file and set the timezone.

**Comment by rallytime** [comment]
Closed via #44605
